ecosystem-modules/starport/src/lib.rs

```diff
@@ -0,0 +1,433 @@
+
+		/// The percentage threshold of authorities signatures required for Notices to take effect.
+		type PercentThresholdForAuthoritySignature: Get<Perbill>;
+
+		/// The pallet handling Compound's Cash tokens
+		type Cash: CompoundCashTrait<Balance, Moment>;
+	}
+
+	#[pallet::error]
+	pub enum Error<T> {
+		/// Acala -> Compound Gateway
+		/// There are not enough supply on the Compound chain for the lock operation.
+		InsufficientAssetSupplyCap,
+
+		/// Notices from Compound
+		/// The same notice cannot be invoked more than once.
+		NoticeAlreadyInvoked,
+		/// Only specific Admin account is able to send Notice to be invoked
+		InvalidNoticeInvoker,
+		/// The Admin account does not have enough asset for the Unlock operation.
+		InsufficientAssetToUnlock,
+		/// Not enough authorities have signed this notice for it to be effective.
+		InsufficientValidNoticeSignatures,
+		/// Too many Authorities.
+		ExceededMaxNumberOfAuthorities,
+		/// Authorities cannot be empty
+		AuthoritiesListCannotBeEmpty,
+		/// Only the current Gateway Admin can change the current Gateway Admin.
+		InvalidGatewayAdminUpdateCaller,
+	}
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(crate) fn deposit_event)]
+	#[pallet::metadata(Balance = "Balance", T::AccountId = "AccountId")]
+	pub enum Event<T: Config> {
+		/// User has locked some asset and uploaded them into Compound.
+		AssetLockedTo(CurrencyId, Balance, T::AccountId),
+
+		/// The user has unlocked some asset and downloaded them back into Acala.
+		AssetUnlocked(CurrencyId, Balance, T::AccountId),
+
+		/// The list of authorities has been updated.
+		GatewayAuthoritiesChanged,
+
+		/// The supply cap for an asset has been updated.
+		SupplyCapSet(CurrencyId, Balance),
+
+		/// The future yield for CASH is set.
+		FutureYieldSet(Balance, CashYieldIndex, Moment),
+
+		/// Gateway's Admin Account has been updated.
+		GatewayAdminUpdated(T::AccountId),
+	}
+
+	#[derive(Encode, Decode, Clone, RuntimeDebug, PartialEq, Eq)]
+	pub struct GatewayNotice<AccountId> {
+		pub id: u64,
+		pub payload: GatewayNoticePayload<AccountId>,
+	}
+
+	impl<AccountId> GatewayNotice<AccountId> {
+		pub fn new(id: u64, payload: GatewayNoticePayload<AccountId>) -> Self {
+			Self { id, payload }
+		}
+	}
+
+	#[derive(Encode, Decode, Clone, RuntimeDebug, PartialEq, Eq)]
+	pub enum GatewayNoticePayload<AccountId> {
+		/// Update the current supply cap for an asset. Only assets that have spare supplies.
+		/// can be locked or uploaded to the Compound chain.
+		SetSupplyCap(CurrencyId, Balance),
+
+		/// Update the current set of authorities who sign Notices.
+		ChangeAuthorities(Vec<CompoundAuthoritySignature>),
+
+		/// Unlock or download assets from Compound chain back into Acala chain.
+		Unlock(CurrencyId, Balance, AccountId),
+
+		/// Set the future yield for the Cash asset.
+		/// Parameters: uint128 nextCashYield, uint128 nextCashYieldIndex, uint nextCashYieldStart
+		SetFutureYield(Balance, CashYieldIndex, Moment),
```

Comment:
```suggestion
		SetFutureYield { next_cash_yield: Balance, next_cash_yield_index: CashYieldIndex, next_cash_yield_start: Moment },
```

Prefer to use named arguments if there are multiple arguments